Added option to disable startup splash per issue #13

robfrawley opened the pull request:
This PR resolves #13 by adding the option to define `SH110X_NO_SPLASH` to disable the startup splash image shown by both the `Adafruit_SH1106G::begin()` and `Adafruit_SH1107::begin()` method implementations (this was done using the exact same implementation found in the related [Adafruit SSD1306 library](https://github.com/adafruit/Adafruit_SSD1306)).

I am hard-pressed to imagine how these changes could cause any limitations or regressions in this library and have tested these changes using the provided examples and my own codebase without any issues. I won't go into any further details about the library code changes, as it should be exceptionally straightforward and easy to review the 6 lines of code this PR introduces ;-).

Please advise if there is anything I much change for this PR to be merged.

robfrawley committed:
add option to disable startup splash per request in issue #13

robfrawley commented:
Looks like your CI clang build test suite is broken somehow, but I don't believe my changes are related in any way... Please advise if you determine otherwise.

ladyada commented:
hiya! thanks so much for submitting a PR! we can review & merge PRs once they have passed continuous integration (CI). that means that code compiles cleanly for all tested platforms, is clang formatted so we maintain the same text formatting for all new code, and is doxygen documented.
if your code isnt passing, check the CI output (click on the red X next to the PR to scroll through the log and find where the error is

here is a tutorial on doxygen: https://learn.adafruit.com/the-well-automated-arduino-library/doxygen

and clang-format: https://learn.adafruit.com/the-well-automated-arduino-library/formatting-with-clang-format

and overall how to contribute PRs to libraries: https://learn.adafruit.com/contribute-to-arduino-with-git-and-github

once you get that green checkmark that indicates CI has passed, please comment reply to this post so we know its time for another review (we may not get notified on CI pass and miss that its time to look!)

robfrawley commented:
@ladyada: Oops, I thought it was a clang compilation error (due to the "clang" header in the Github actions log) and couldn't fathom how any of my additions would have caused such an issue, but I now see it was a failure related to clang-format. ;-) I've fixed the code style via clang-format and rebased my two commits into a single one; should be good to merge now! Thanks for the documentation links.

ladyada commented:
yay! great work! :)
